api/src/pcapi/routes/pro/public_api/individual/endpoints.py

```diff
@@ -0,0 +1,108 @@
+import logging
+
+from pcapi.core.offerers import models as offerers_models
+from pcapi.core.offers import api as offers_api
+from pcapi.core.offers import exceptions as offers_exceptions
+from pcapi.models import api_errors
+from pcapi.serialization.decorator import spectree_serialize
+from pcapi.validation.routes.users_authentifications import api_key_required
+from pcapi.validation.routes.users_authentifications import current_api_key
+
+from . import blueprint
+from . import serialization
+
+
+logger = logging.getLogger(__name__)
+
+PRODUCT_OFFERS_TAG = "Product offers"
+
+
+def _retrieve_venue_from_location(
+    location: serialization.DigitalLocation | serialization.PhysicalLocation,
+) -> offerers_models.Venue:
+    offerer_id = current_api_key.offererId  # type: ignore[attr-defined]
+    if isinstance(location, serialization.PhysicalLocation):
+        venue: offerers_models.Venue | None = (
+            offerers_models.Venue.query.join(offerers_models.Offerer)
+            .filter(
+                offerers_models.Venue.id == location.venue_id,
+                offerers_models.Offerer.id == offerer_id,
+            )
+            .first()
```

Comment:
```suggestion
            offerers_models.Venue.query
            .filter_by(id=location.venue_id, managingOffererId=offerer_id)
            .one_or_none()
```

idem plus bas dans la branche `else`.

Au passage, je propose d'utiliser `one_or_none()`. C'est équivalent à `first()`, mais c'est plus clair qu'on n'attend qu'un seul élément ou zéro. (C'est un détail.)